File: im_lib/src/main/java/vip/qsos/im/lib/model/HeartbeatRequest.java
package vip.qsos.im.lib.model;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

import vip.qsos.im.lib.constant.CIMConstant;

/**
 * 服务端心跳请求
 */
public class HeartbeatRequest implements Serializable, Protobufable {

    private static final long serialVersionUID = 1L;
    private static final String TAG = "SERVER_HEARTBEAT_REQUEST";
    private static final String CMD_HEARTBEAT_REQUEST = "SR";

    private static HeartbeatRequest object = new HeartbeatRequest();

    private HeartbeatRequest() {

    }

    public static HeartbeatRequest getInstance() {
        return object;
    }

    @Override
    public byte[] getByteArray() {
        return CMD_HEARTBEAT_REQUEST.getBytes();
    }

    @Override
    @NotNull
    public String toString() {
        return TAG;
    }

    @Override
    public byte getType() {
        return CIMConstant.ProtobufType.Companion.getS_H_RQ();
    }

}

File: im_lib/src/main/java/vip/qsos/im/lib/model/HeartbeatRequest.kt
package vip.qsos.im.lib.model

import vip.qsos.im.lib.constant.IMConstant

/**
 * @author : 华清松
 * 服务端心跳请求实体
 */
class HeartbeatRequest private constructor() : IProtobufAble {

    companion object {
        private const val serialVersionUID = 1L
        const val TAG = "SERVER_HEARTBEAT_REQUEST"
        const val SERVER_HEARTBEAT_REQUEST = "SR"
        val instance = HeartbeatRequest()
    }

    override val byteArray: ByteArray
        get() = SERVER_HEARTBEAT_REQUEST.toByteArray()

    override val type: Byte
        get() = IMConstant.ProtobufType.HEART_RQ

    override fun toString(): String {
        return TAG
    }

}
